octoprint_mrbeam/static/js/convert.js

```diff
@@ -616,25 +634,53 @@ $(function () {
             self._update_job_summary();
         });
 
+        self.showMaterialCompatibility = function () {
+            let customMaterialsLaserModels = [];
+            let customMaterials = self.custom_materials();
+            // Get all laser models used while saving custom materials
+            for (let materialKey in customMaterials) {
+                if (customMaterials.hasOwnProperty(materialKey)) {
+                    let m = customMaterials[materialKey];
+                    if (('laser_model' in m) && !customMaterialsLaserModels.includes(m.laser_model)) {
+                        customMaterialsLaserModels.push(m.laser_model);
+                    }
+                }
+            }
+            return customMaterialsLaserModels.some(item => item !== MRBEAM_LASER_HEAD_MODEL);
+        }
+
         self.filterQuery = ko.observable("");
         self.filteredMaterials = ko.computed(function () {
             // just to subscribe to this obserable!
             self.material_settings2_updated_trigger();
 
-            var q = self.filterQuery();
-            var out = [];
+            let q = self.filterQuery();
+            let out = [];
             // List custom materials first
             // filter custom materials
-            var customs = self.custom_materials();
-            for (var materialKey in customs) {
-                var m = customs[materialKey];
+            let customs = self.custom_materials();
+            // Show material compatibility when different laserhead models are detected
+            let materialCompatibilityDisplay = self.showMaterialCompatibility();
+            for (let materialKey in customs) {
+                let m = customs[materialKey];
                 if (m !== null) {
                     //					m.name = materialKey; // TODO i18n
                     if (m.name.toLowerCase().indexOf(q) >= 0) {
                         m.key = materialKey;
                         m.custom = true;
                         out.push(m);
                     }
+                    if (materialCompatibilityDisplay) {
+                        m.compatible = m.laser_model === MRBEAM_LASER_HEAD_MODEL;
+                        if (m.laser_model === 'S') {
+                            m.customBeforeElementContent = '[S]';
+                        }
```

Comment:
No this has been clarified and added in the design ticket. The [S] symbol will be showing only when the S model is detected in the custom materials. It can be compatible or not